core/client/db/src/lib.rs

```diff
@@ -453,6 +451,34 @@ impl<Block: BlockT> Backend<Block> {
 
 		Ok(())
 	}
+
+	// write stuff to a transaction after a new block is finalized.
+	// this manages state pruning. Fails if called with a block which
+	// was not a child of the last finalized block.
```

Comment:
The comment is misleading. State pruning is managed by `state-db`. This only manages canonicalization.